src/libraries/Native/Unix/System.Globalization.Native/pal_icushim.c

```diff
@@ -374,7 +374,10 @@ int32_t GlobalizationNative_LoadICU()
 // return the current loaded ICU version
 int32_t GlobalizationNative_GetICUVersion()
 {
-    int32_t version;
-    u_getVersion((uint8_t *) &version);
+    int32_t version = 0;
```

Comment:
This would avoid the dirty cast to `uint8_t *`, and also make the IsLittleEndian check unnecessary at the place where this is called:

```
UVersionInfo versionInfo;

if (u_getVersion_ptr == NULL)
    return 0;

u_getVersion(&versionInfo);

return (versionInfo[0] << 24) + (versionInfo[1] << 16) + (versionInfo[2] << 8) + versionInfo[3];
```

Reply:
Nice 😄

Reply:
It seems like even after this we either need versionInfo to be a uint_8* or cast it. I'm getting:

```
'void (uint8_t *)': cannot convert argument 1 from 'UVersionInfo (*)
' to 'uint8_t *'
```

is that maybe because of the way we define the _ptr for the function with the macros and UVersionInfo is getting replaced with uint8_t?

This is what icu.h has:
```c
typedef uint8_t UVersionInfo[U_MAX_VERSION_LENGTH];
u_getVersion(UVersionInfo versionArray);
```

I guess I'm going to declare my array using `[U_MAX_VERSION_LENGTH];` instead.

Reply:
I think replacing `u_getVersion(&versionInfo);` -> `u_getVersion(versionInfo);` in my suggestion should fix this.

Reply:
Ah right, I think that should fix it cause it is already a ptr. I'll try it out.